src/lib/Menu/LeftSidebarBuilder.php

```diff
@@ -21,27 +21,33 @@
  */
 class LeftSidebarBuilder extends AbstractBuilder implements TranslationContainerInterface
 {
-    /** @var ConfigResolverInterface */
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
     private $configResolver;
 
+    /** @var \EzSystems\EzPlatformAdminUiBundle\Templating\Twig\UniversalDiscoveryExtension */
+    private $udwExtension;
+
     /* Menu items */
     const ITEM__SEARCH = 'sidebar_left__search';
     const ITEM__BROWSE = 'sidebar_left__browse';
     const ITEM__TRASH = 'sidebar_left__trash';
 
     /**
-     * @param MenuItemFactory $factory
-     * @param EventDispatcherInterface $eventDispatcher
-     * @param ConfigResolverInterface $configResolver
+     * @param \EzSystems\EzPlatformAdminUi\Menu\MenuItemFactory $factory
+     * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $eventDispatcher
+     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
+     * @param \EzSystems\EzPlatformAdminUiBundle\Templating\Twig\UniversalDiscoveryExtension $udwExtension
      */
     public function __construct(
         MenuItemFactory $factory,
         EventDispatcherInterface $eventDispatcher,
-        ConfigResolverInterface $configResolver
+        ConfigResolverInterface $configResolver,
+        UniversalDiscoveryExtension $udwExtension
     ) {
         parent::__construct($factory, $eventDispatcher);
 
         $this->configResolver = $configResolver;
+        $this->udwExtension = $udwExtension;
```

Comment:
As we discussed - can you move the logic out of `UniversalDiscoveryExtension` and use it in both places?